Modules/_testexternalinspection.c

```diff
@@ -1464,6 +1471,240 @@ find_running_task(
     return 0;
 }
 
+static int
+append_awaited_by_for_thread(
+    int pid,
+    uintptr_t head_addr,
+    struct _Py_DebugOffsets *debug_offsets,
+    struct _Py_AsyncioModuleDebugOffsets *async_offsets,
+    PyObject *result
+) {
+    struct llist_node task_node;
+
+    if (0 > read_memory(
+                pid,
+                head_addr,
+                sizeof(task_node),
+                &task_node))
+    {
+        return -1;
+    }
+
+    while ((uintptr_t)task_node.next != head_addr) {
+        uintptr_t task_addr = (uintptr_t)task_node.next
```

Comment:
Unless I am missing something this loop control is dangerous. You're accessing `task_node.next` after the loop condition check, but without validating that it's not NULL. Maybe we should be adding a null check before the memory read.

Reply:
oh good one